pytest_rerunfailures.py

```diff
@@ -132,7 +132,9 @@ def _remove_cached_results_from_fixtures(item):
         fixture_defs = fixture_info.name2fixturedefs[fixture_def_str]
         for fixture_def in fixture_defs:
             if hasattr(fixture_def, cached_result):
-                delattr(fixture_def, cached_result)
+                result, cache_key, err = getattr(fixture_def, cached_result)
+                if err:  # Deleting cached results for only failed fixtures
+                    delattr(fixture_def, cached_result)
```

Comment:
I like the test for an error. Hopefully we will not have a chain of dependent fixtures and strange results, but then the dependent fixture should also produce an error.

Reply:
I've added the test for the error (test_rerun_on_module_fixture_with_reruns). Only setup_fixture in that scenario will fail and will be cleaned up. Dependent fixture cannot produce any error if one of the fixtures before fails.
Unless I am missing something.